dev/integration_tests/channels/lib/src/basic_messaging.dart

```diff
@@ -78,6 +79,37 @@ Future<TestStepResult> basicStandardHandshake(dynamic message) async {
       'Standard >${toString(message)}<', channel, message);
 }
 
+Future<void> _basicBackgroundStandardEchoMain(List<Object> args) async {
+  final SendPort sendPort = args[2] as SendPort;
+  final Object message = args[1];
+  final String name = 'Background Echo >${toString(message)}<';
+  const String description =
+      'Uses a platform channel from a background isolate.';
+  try {
+    BackgroundIsolateBinaryMessenger.ensureInitialized(
+        args[0] as RootIsolateToken);
+    const BasicMessageChannel<dynamic> channel = BasicMessageChannel<dynamic>(
+      'std-echo',
+      ExtendedStandardMessageCodec(),
+    );
+    final Object response = await channel.send(message) as Object;
+
+    final TestStatus testStatus = TestStepResult.deepEquals(message, response)
+        ? TestStatus.ok
+        : TestStatus.failed;
+    sendPort.send(TestStepResult(name, description, testStatus));
+  } catch (ex) {
+    sendPort.send(TestStepResult(name, description, TestStatus.failed,
+        error: ex.toString()));
+  }
+}
+
+Future<TestStepResult> basicBackgroundStandardEcho(Object message) async {
+  final ReceivePort receivePort = ReceivePort();
+  Isolate.spawn(_basicBackgroundStandardEchoMain, <Object>[RootIsolateToken.instance!, message, receivePort.sendPort]);
```

Comment:
FWIW, as a general rule, in the root isolate we should never access dart:ui directly, only via the bindings. That's why in my branch I indirected via the binding to get the RootIsolateToken. Otherwise we have no way to mock dart:ui in the tests when we need to. (That's also why I had RootIsolateToken be a "plain old data" class with the logic to get one on the PlatformDispatcher; the "plain old data" classes can be used directly in tests, and the PlatformDispatcher is mockable, but class statics cannot and are not.)

Reply:
Makes sense why you'd put it on the `ServiceBinding`, did it.

I don't know what you mean by "plain old data".  Your patch didn't have the changes to the engine so it's hard to see what you mean.

Reply:
I had to introduce "useBackgroundIsolateBinaryMessenger" to make the existing mocks work.  Since we have test coverage with integration tests I'm not sure if we'll ever have to mock the GetRootIsolateToken.